validator/client/attest_protect.go

```diff
@@ -66,37 +75,51 @@ func (v *validator) postAttSignUpdate(ctx context.Context, indexedAtt *ethpb.Ind
 
 // isNewAttSlashable uses the attestation history to determine if an attestation of sourceEpoch
 // and targetEpoch would be slashable. It can detect double, surrounding, and surrounded votes.
-func isNewAttSlashable(history *slashpb.AttestationHistory, sourceEpoch, targetEpoch uint64) bool {
+func isNewAttSlashable(ctx context.Context, history *kv.EncHistoryData, sourceEpoch, targetEpoch uint64, signingRoot [32]byte) bool {
 	if history == nil {
 		return false
 	}
-	farFuture := params.BeaconConfig().FarFutureEpoch
 	wsPeriod := params.BeaconConfig().WeakSubjectivityPeriod
-
 	// Previously pruned, we should return false.
-	if int(targetEpoch) <= int(history.LatestEpochWritten)-int(wsPeriod) {
+	lew, err := history.GetLatestEpochWritten(ctx)
+	if err != nil {
+		log.WithError(err).Error("Could not get latest epoch written from encapsulated data")
+		return false
+	}
+	if int(targetEpoch) <= int(lew)-int(wsPeriod) {
```

Comment:
is it safe to downcast this? We should keep uint64 and instead do an overflow check

Reply:
thanks